### DISCUSSION TITLE
2.0中field上的value类型报错

### DISCUSSION THREAD

**Discussion by aspirantzhang** [alibaba/formily#1257] [Q&A 2.x] [answered]
![01.png](https://i.loli.net/2021/04/20/ShmfsIbQ2V3LTkg.png)
![02.jpg](https://i.loli.net/2021/04/20/tzYVdeiu6WmIoCF.jpg)

让value这东西没在field上，这个未知应该咋写捏
https://codesandbox.io/s/formily-value-type-u84rc?file=/App.tsx

**Comment by janryWang** [comment, marked as answer]
需要依赖type checker做类型守护，isField

**Comment by aspirantzhang** [comment]
解决了

```
import { isField } from '@formily/core';

...
    onFieldReact('*.*.uri', (field) => {
      if (isField(field)) {  //+
        field.value = field.value?.replace('admins', field.query('routeName').get('value'));
      }
    });
```
https://core.formilyjs.org/api/entry/form-checker
